### ISSUE TITLE
Bad voice after Fine-tuning the pre-trained weights of Hindi model

### ISSUE THREAD

**Issue by shekharmeena2896** [SWivid/F5-TTS#1058]
### Checks

- [x] This template is only for usage issues encountered.
- [x] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [x] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [x] I am using English to submit this issue to facilitate community communication.

### Environment Details

I used train.py file to finetune the hindi pretrained weights, below is my hindi.yaml configuration :

hydra:
  run:
    dir: ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}/${now:%Y-%m-%d}/${now:%H-%M-%S}

datasets:
  name: hindi_dataset_custom
  batch_size_per_gpu: 3200
  batch_size_type: frame
  max_samples: 64
  num_workers: 8  # Reduced from 16 to be safe

optim:
  epochs: 200
  learning_rate: 1e-5
  num_warmup_updates: 500
  grad_accumulation_steps: 1
  max_grad_norm: 1.0
  bnb_optimizer: False

model:
  name: F5TTS_Hindi_Finetune
  tokenizer: custom  # Using custom for Hindi
  tokenizer_path: /home/ubuntu/F5-TTS/data/hindi_dataset_custom/vocab.txt  # Path to your Hindi vocab
  backbone: DiT  # This was missing in your config!
  arch:
    dim: 768
    depth: 18
    heads: 12
    ff_mult: 2
    text_dim: 512
    text_mask_padding: False
    conv_layers: 4
    pe_attn_head: 1
    checkpoint_activations: True
  mel_spec:
    target_sample_rate: 24000
    n_mel_channels: 100
    hop_length: 256
    win_length: 1024
    n_fft: 1024
    mel_spec_type: vocos
  vocoder:
    is_local: False
    local_path: null

ckpts:
  logger: wandb  # Set to null for simpler setup, or use wandb/tensorboard if desired
  log_samples: True
  save_per_updates: 5000  # Reduced from 50000 for more frequent saves
  keep_last_n_checkpoints: 3  # Keep only the last 3 checkpoints to save space
  last_per_updates: 500
  save_dir: ckpts/${model.name}_vocos_custom_hindi_dataset
  checkpoint_path: /home/ubuntu/F5-TTS/F5-Hindi-24KHz/model_2500000.pt

### Steps to Reproduce

 I changed the path of of the kept in the train.py as suggested in #16 but there exist not test_train.py so I used train.py to finetune and changed the directory the pretrained model in checkpoint_path

Hi I was trying to finetune the pretrained weights of hindi model, I changed the path in train.py (checkpoint_paths to the pretrained directory as suggested by @SWivid and my config file is below and there existed no test_train.py, assuming I used train.py

hydra:
run:
dir: ckpts/${model.name}${model.mel_spec.mel_spec_type}${model.tokenizer}_${datasets.name}/${now:%Y-%m-%d}/${now:%H-%M-%S}

datasets:
name: hindi_dataset_custom
batch_size_per_gpu: 3200
batch_size_type: frame
max_samples: 64
num_workers: 8 # Reduced from 16 to be safe

optim:
epochs: 500
learning_rate: 1e-5
num_warmup_updates: 500
grad_accumulation_steps: 1
max_grad_norm: 1.0
bnb_optimizer: False

model:
name: F5TTS_Hindi_Finetune
tokenizer: custom # Using custom for Hindi
tokenizer_path: /home/ubuntu/F5-TTS/data/hindi_dataset_custom/vocab.txt # Path to your Hindi vocab
backbone: DiT # This was missing in your config!
arch:
dim: 768
depth: 18
heads: 12
ff_mult: 2
text_dim: 512
text_mask_padding: False
conv_layers: 4
pe_attn_head: 1
checkpoint_activations: True
mel_spec:
target_sample_rate: 24000
n_mel_channels: 100
hop_length: 256
win_length: 1024
n_fft: 1024
mel_spec_type: vocos
vocoder:
is_local: False
local_path: null

ckpts:
logger: wandb # Set to null for simpler setup, or use wandb/tensorboard if desired
log_samples: True
save_per_updates: 5000 # Reduced from 50000 for more frequent saves
keep_last_n_checkpoints: 3 # Keep only the last 3 checkpoints to save space
last_per_updates: 500
save_dir: ckpts/my_hindi_finetune_run # Path to your pretrained model
(f5tts) root@t1-le-45-gra7:/home/ubuntu/F5-TTS/src/f5_tts/configs#

the training finish instantly I am not able to understand what mistake I am making this below is the output :

(f5tts) root@t1-le-45-gra7:/home/ubuntu/F5-TTS/src/f5_tts/configs# cd ../../..
(f5tts) root@t1-le-45-gra7:/home/ubuntu/F5-TTS# tail -f nohup.out
Loading dataset ...
Download Vocos from huggingface charactr/vocos-mel-24khz
Sorting with sampler... if slow, check whether dataset is provided with duration: 100%|██████████| 5900/5900 [00:00<00:00, 1259166.21it/s]
Creating dynamic batches with 3200 audio frames per gpu: 100%|██████████| 5900/5900 [00:00<00:00, 1693798.33it/s]
wandb:
wandb: 🚀 View run F5TTS_Hindi_Finetune_vocos_custom_hindi_dataset_custom at: https://wandb.ai/devteeoff-teeoff-technologies/CFM-TTS/runs/vtuhvwdi
wandb: ⭐️ View project at: https://wandb.ai/devteeoff-teeoff-technologies/CFM-TTS
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
wandb: Find logs at: ./wandb/run-20250527_073442-vtuhvwdi/logs
Saved last checkpoint at update 2500000

**Comment by SWivid** [comment]
As you mentioned "bad voice", but there's no wav files attached?
Have you done pretraining or finetuning, if finetuning, what is the pretrained model source (repo link e.g.)? And how do you organize your data and training process?
It seems like you are working on a different codebase other than us, so are there any specific changes?

**Comment by shekharmeena2896** [comment]
so I am using spring labs pretrained hindi model which is provided in the inference guide, I collected the high quality female data single speaker about 13 hours of wav file converted into required format:
metadata.csv and wavs, metadata.csv contains audio|text and then used  [prepare_csv_wavs.py](https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/train/datasets/prepare_csv_wavs.py) to make it in the below format  : 
(f5tts) root@t1-le-45-gra7:/home/ubuntu/F5-TTS/data# ls hindi_dataset_custom/
duration.json  raw.arrow  vocab.txt.


and then I changed the path in train.py the checkpoint_path = to the pretrained model which is hindi model by springs lab which is mentioned in inference.md 


and this is the config file which I used to finetune :
hydra:
  run:
    dir: ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}/${now:%Y-%m-%d}/${now:%H-%M-%S}

datasets:
  name: hindi_dataset_custom
  batch_size_per_gpu: 3200
  batch_size_type: frame
  max_samples: 64
  num_workers: 8  # Reduced from 16 to be safe

optim:
  epochs: 500
  learning_rate: 1e-5
  num_warmup_updates: 500
  grad_accumulation_steps: 1
  max_grad_norm: 1.0
  bnb_optimizer: False

model:
  name: F5TTS_Hindi_Finetune
  tokenizer: custom  # Using custom for Hindi
  tokenizer_path: /home/ubuntu/F5-TTS/data/hindi_dataset_custom/vocab.txt  # Path to your Hindi vocab
  backbone: DiT  # This was missing in your config!
  arch:
    dim: 768
    depth: 18
    heads: 12
    ff_mult: 2
    text_dim: 512
    text_mask_padding: False
    conv_layers: 4
    pe_attn_head: 1
    checkpoint_activations: True
  mel_spec:
    target_sample_rate: 24000
    n_mel_channels: 100
    hop_length: 256
    win_length: 1024
    n_fft: 1024
    mel_spec_type: vocos
  vocoder:
    is_local: False
    local_path: null

ckpts:
  logger: wandb  # Set to null for simpler setup, or use wandb/tensorboard if desired
  log_samples: True
  save_per_updates: 5000  # Reduced from 50000 for more frequent saves
  keep_last_n_checkpoints: 3  # Keep only the last 3 checkpoints to save space
  last_per_updates: 500
  save_dir: ckpts/my_hindi_finetune_run  # Path to your pretrained model




and I ran this command  to fine the hindi model on custom voice:

accelerate launch --mixed_precision=fp16 src/f5_tts/train/train.py --config-name F5TTS_Small.yaml

**Comment by SWivid** [comment]
as you are practicing finetuning, have you use the save vocab.txt as theirs?
